Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx

```diff
@@ -0,0 +1,654 @@
+    setLocalRootLuisRegion(opt.region);
+  };
+  const onChangeLUISPrediction = async (_, opt) => {
+    // get list of luis keys for this subscription
+    setLocalRootLuisEndpointKey(opt.key);
+  };
+
+  const onChangeAction = async (_, opt) => {
+    setNextAction(opt.key);
+  };
+
+  const onChangeResourceGroup = async (_, opt) => {
+    setResourceGroupKey(opt.key);
+    setResourceGroup(opt.text);
+    if (opt.key === CREATE_NEW_KEY) {
+      setCreateResourceGroup(true);
+    } else {
+      setCreateResourceGroup(false);
+    }
+  };
+
+  const performNextAction = () => {
+    if (nextAction === 'choose') {
+      // close the modal!
+      props.onGetKey({
+        authoringKey: localRootLuisKey,
+        endpointKey: localRootLuisEndpointKey,
+        authoringRegion: localRootLuisRegion,
+      });
+      setOutcomeDescription(formatMessage('The following LUIS keys have been successfully added to your bot project:'));
+      setOutcomeSummary(
+        <div>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Authoring key')}</label>
+            {localRootLuisKey}
+          </p>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Endpoint Key')}</label>
+            {localRootLuisEndpointKey}
+          </p>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Region')}</label>
+            {localRootLuisRegion}
+          </p>
+        </div>
+      );
+      setOutcomeError(false);
+
+      setCurrentPage(3);
+    } else if (nextAction === 'handoff') {
+      setShowHandoff(true);
+      props.onDismiss();
+    } else {
+      setCurrentPage(2);
+    }
+  };
+
+  const iconStyles = { marginRight: '8px' };
+  const onRenderOption = (option) => {
+    return (
+      <div>
+        {option.data && option.data.icon && (
+          <Icon aria-hidden="true" iconName={option.data.icon} style={iconStyles} title={option.data.icon} />
+        )}
+        <span>{option.text}</span>
+      </div>
+    );
+  };
+
+  return (
+    <Fragment>
+      {showAuthDialog && (
+        <AuthDialog
+          needGraph={false}
+          next={hasAuth}
+          onDismiss={() => {
+            setShowAuthDialog(false);
+          }}
+        />
+      )}
+      <ProvisionHandoff
+        developerInstructions={formatMessage('Send this to your IT admin')}
+        handoffInstructions={handoffInstructions}
+        hidden={!showHandoff}
+        title={formatMessage('Generate a provisioning request')}
+        onDismiss={() => setShowHandoff(false)}
+      />
+      <Dialog
+        dialogContentProps={{
+          type: DialogType.normal,
+          title: currentPage === 2 ? formatMessage('Create new LUIS resources') : formatMessage('Select LUIS keys'),
+        }}
+        hidden={props.hidden}
+        minWidth={480}
+        modalProps={{
+          isBlocking: false,
+        }}
+        onDismiss={props.onDismiss}
+      >
+        {currentPage === 1 && (
```

Comment:
Nit: Consider methods for rendering each page and then a method that switches off the current page number.  It helps it not be a such a long and indented chunk of UI.

Reply:
good idea! done.